### PR TITLE
Fix #246: Error when use provider_name = TENANT on 0.6.0 mongodbatlas provider version

### LINKED ISSUE #246: Error when use provider_name = TENANT on 0.6.0 mongodbatlas provider version.
This error only occurs when i user 0.6.0 version from mongodbatlas provider, on 0.5.1 works fine.

```sh
Error: error reading MongoDB Cluster (mongoexamplecluster): GET https://cloud.mongodb.com/api/atlas/v1.0/groups/5ee7f662eae0c80ddc83b3f0/containers?providerName=TENANT: 404 (request "Not Found") Provider TENANT not currently supported.

  on ../main.tf line 9, in resource "mongodbatlas_cluster" "cluster":
   9: resource "mongodbatlas_cluster" "cluster" {
```
To reproduce you can use TENANT example. 

```
resource "mongodbatlas_cluster" "cluster-test" {
  project_id              = "<YOUR-PROJECT-ID>"
  name                    = "cluster-test-global"
  //M2 must be 2, M5 must be 5
  disk_size_gb            = "2"

  //Provider Settings "block"
  provider_name = "TENANT"
  backing_provider_name = "AWS"
  provider_region_name = "US_EAST_1"
  provider_instance_size_name = "M2"

  //These must be the following values
  mongo_db_major_version = "4.2"
  auto_scaling_disk_gb_enabled = "false"
}
```

### PR DESCRIPTION
**Pull request by PacoDw**
## Description

- Fixed the cluster TENANT error
- Fixed the `provider_encrypt_ebs_volume` attribute it had a state error
- Formatted and fixed the test cases of the cluster

Link to any related issue(s): #246 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the MongoDB CLA
- [x] I have read the Terraform contribution guidelines 
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_cluster.go [hunk 1]
```diff
@@ -711,14 +711,16 @@ func resourceMongoDBAtlasClusterRead(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf(errorClusterSetting, "labels", clusterName, err)
 	}
 
-	containers, _, err := conn.Containers.List(context.Background(), projectID,
-		&matlas.ContainersListOptions{ProviderName: providerName})
-	if err != nil {
-		return fmt.Errorf(errorClusterRead, clusterName, err)
-	}
+	if providerName != "TENANT" {
+		containers, _, err := conn.Containers.List(context.Background(), projectID,
```
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_cluster.go:715]: oh it was the container id!

**Review by themantissa — APPROVED** [review]
LGTM - the container id definitely had some side effects!  Thank you for finding quickly.  Ensure we have one more DoU review to be sure.

**Comment by themantissa** [comment]
One of the issue reporters tested and verified fix works.

File: mongodbatlas/resource_mongodbatlas_cluster.go (hunk 1 above)
Comment (PacoDw) [reply mongodbatlas/resource_mongodbatlas_cluster.go:715]: yes it is :S

**Comment by PacoDw** [comment]
I will merge it even with the Travis error because that error is referring to use the.GetOkExist()` function part of Terraform so we need this function, Travis has a wrong lint that causes it.

**Review by gmlp — APPROVED** [review]
LGTM

**Pull request merged**